adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs

```diff
@@ -119,6 +120,15 @@ public AuthenticationContext(string authority, bool validateAuthority, TokenCach
             _wsTrustWebRequestManager = new WsTrustWebRequestManager(_httpManager);
         }
 
+        /// <summary>
+        /// Provides a mechanism for adding custom message handlers in the HttpClient used by this library.
+        /// </summary>
+        public void UseCustomHttpMessageHandler(HttpMessageHandler httpMessageHandler)
```

Comment:
This is the public API change. I don't want to pass it through a ctor, because we already have a ton of ctors, but can do.

Reply:
The concern I have with this is technically the user could change this at any time and we will likely have passed the _httpManager around since it's a dependency injection and this effectively changing the graph.  We should add this as an optional to the constructor(s) instead of this or we'll likely need more work to ensure this gets used consistently.

Better API for this (to think about for msal, not here) would be to have a builder or the config object for setting this up.